cmd/sourced/compose/workdir/workdir.go

```diff
@@ -0,0 +1,153 @@
+package workdir
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	goerrors "gopkg.in/src-d/go-errors.v1"
+
+	datadir "github.com/src-d/sourced-ce/cmd/sourced/dir"
+)
+
+const activeDir = "__active__"
+
+var (
+	// RequiredFiles list of required files in a directory to treat it as a working directory
+	RequiredFiles = []string{".env", "docker-compose.yml"}
+
+	// OptionalFiles list of optional files that could be deleted when pruning
+	OptionalFiles = []string{"docker-compose.override.yml"}
+
+	// ErrMalformed is the returned error when the workdir is wrong
+	ErrMalformed = goerrors.NewKind("workdir %s is not valid: %s")
+)
+
+type WorkdirType int
+
+const (
+	Local WorkdirType = iota
+	Orgs
+)
+
+type Workdir struct {
+	Type WorkdirType
+	Name string
+	Path string
+}
+
+type builder struct {
+	workdirsPath string
+}
+
+func (b *builder) build(path string) (*Workdir, error) {
```

Comment:
let's add some documentation. Does it expect absolute path or relative to `workdirsPath`?

Reply:
Done [here](https://github.com/src-d/sourced-ce/pull/183/commits/0977d1e86bea186ff23f12a43219c8537f82380b).